sync with the 64bit repo

haubi opened the pull request.

haubi committed:
install 'gcc-g++' host package

When we find host gcc version newer than gcc-4.8 we really expect the
C++ compiler to be available as well.

haubi committed:
use latest tree and portage version

to make it a real CI build rather than using old snapshots

haubi committed:
drop ENV CHOST=i686-pc-linux-gnu, use linux32 instead

awesomebytes commented on initial_bootstrap/Dockerfile:
```diff
@@ -32,11 +32,7 @@ RUN cat /proc/cpuinfo && cat /proc/meminfo && df -h
 ENV EPREFIX /tmp/gentoo
 
-# Needed for 32b
-ENV CHOST i686-pc-linux-gnu
```
Without this it won't bootstrap as 32bit, please add that back

awesomebytes replied:
@haubi

haubi replied:
Well, running bootstrap-prefix.sh using that 'linux32' really should be sufficient here.

awesomebytes replied:
I tried that (for a while, with help from the IRC channel) and some packages managed to ignore it somehow and I only found on runtime after building the prefix. By what I remember something still generated 64b binaries/shared libraries. The only way it worked was with the CHOST flag. Can you please bring it back?

I'd give it a second try without it but im currently busy to give it a well thru test.

haubi replied:
Hmm... bootstrap-prefix.sh really should identify CHOST on its own, based on 'uname -m' output:
$ uname -m
x86_64
$ linux32 uname -m
i686

awesomebytes replied:
I know. And that's what others said in the IRC channel. But my experience after trying-to build it, it would fail, or after building it, libraries would fail with incompatible 32-64b problems.

Given it took me a week to get back to you again, I'll just merge it, if it doesn't work as expected, I'll revert it.

awesomebytes commented on initial_bootstrap/Dockerfile.fedora:
```diff
@@ -23,11 +24,8 @@ RUN df -h
 RUN curl -O https://gitweb.gentoo.org/repo/proj/prefix.git/plain/scripts/bootstrap-prefix.sh
 RUN chmod +x bootstrap-prefix.sh
-# Needed for 32b
-ENV CHOST i686-pc-linux-gnu
```
Same here

haubi committed:
using 'linux32' is enough with bootstrap-prefix.sh

haubi committed:
install 32bit system headers

As we're on the 64bit host OS, @development-tools does not provide the
32bit glibc-devel package, hence we need glibc-devel.i686 explicitly.

haubi commented:
@awesomebytes Really, running 'linux32 bootstrap-prefix.sh' is enough to get the 32bit prefix, although that one needs 32bit system libraries (glibc-devel.i686) of course.
The build will proceed until https://bugs.gentoo.org/644372 with this commits.
